src/router/orders/validation.ts

```diff
@@ -57,23 +71,28 @@ export const ordersPost: OrdersPost = async function (req, res, next) {
       throw new ValidationError('Invalid order time');
     }
 
-    const products = await db.products.findMany([
-      FieldPath.documentId(),
-      'in',
-      Object.keys(order.items)
-    ]);
-    const productInStock = products
-      .filter((product) => {
-        if (product.isAvailable) return product;
-      })
-      .map(({ id }) => id);
-    const isProductsValid = Object.keys(order.items).every((id) => {
-      return productInStock.includes(id);
-    });
-    if (!isProductsValid) {
-      throw new ValidationError('Products not found');
+    for (const item of order.items) {
+      const product = await db.products.findOneById(item.productId);
```

Comment:
It looks like you iterates by `order.items` twice. Please move code from the loop above to this loop